boot/bootutil/src/loader.c

```diff
@@ -1261,11 +1268,19 @@ boot_go(struct boot_rsp *rsp)
         slot = 0;
         break;
 
-    case BOOT_SWAP_TYPE_TEST:
-    case BOOT_SWAP_TYPE_PERM:
+    case BOOT_SWAP_TYPE_TEST:          /* fallthrough */
+    case BOOT_SWAP_TYPE_PERM:          /* fallthrough */
     case BOOT_SWAP_TYPE_REVERT:
         slot = 1;
-        boot_set_copy_done();
+        rc = boot_set_copy_done();
+        if (rc != 0) {
+            /* FIXME: this flag is basically used to trigger a revert and
+             * in this specific case a failure would not be fatal and the
+             * image in slot0 could be safely booted, but seems better to
+             * fail now while a better handling is properly thought about.
+             */
+            assert(0);
```

Comment:
Should this be swap_type = BOOT_SWAP_TYPE_PANIC so it happens in production as well as in testing?